repository/src/test/java/com/epam/esm/certificate/CertificateDaoTest.java

```diff
@@ -0,0 +1,123 @@
+package com.epam.esm.certificate;
+
+import com.epam.esm.certificate.dao.CertificateDao;
+import com.epam.esm.certificate.model.Certificate;
+import com.epam.esm.certificate.specification.CertificateSearchSqlBuilder;
+import com.epam.esm.certificate.specification.OrderBy;
+import com.epam.esm.tag.dao.TagDao;
+import org.h2.store.Data;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabase;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType.H2;
+
+public class CertificateDaoTest {
+
+    CertificateDao certificateDao;
+
+    EmbeddedDatabase embeddedDatabase;
+
+    @Before
+    public void init() {
+        embeddedDatabase = new EmbeddedDatabaseBuilder()
+                .generateUniqueName(true)
+                .setType(H2)
+                .setScriptEncoding("UTF-8")
+                .ignoreFailedDrops(true)
+                .addScript("schema.sql")
+                .addScripts("insert-data.sql")
+                .build();
+        certificateDao = new CertificateDao(embeddedDatabase);
+    }
+
+    @Test
+    public void createCertificateTest() {
+        Certificate certificate = new Certificate("name", "description",
+                new BigDecimal("12.6"), 12);
+
+        Certificate expected = certificateDao.create(certificate);
+
+        assertTrue(expected.getId() > 0);
+        assertNotNull(expected.getCreationDate());
+    }
+
+    @Test
+    public void updateCertificateTest() {
```

Comment:
All tests should have structure: given when then
Also, please thunk about naming: methodUnderTest_condition_expectedResult
Moreover, each test should have only one assert